### PR TITLE
Decompile EntityMessageBox

### PR DESCRIPTION
**Pull request by Xeeynamo**
image

Used to display a temporary message that displays a certain information to the player. It needs a new name before getting merged.

### CODE REVIEW

**Review by sozud — APPROVED** [review]
Looks good to me, any of the names suggested in discord are fine

File: include/entity.h [hunk 1]
```diff
@@ -134,6 +134,20 @@ typedef struct {
...
+    s16 unk84;
+    u16 duration;
+    u32 unk88;
+    s16 unk8C;
+    u16 unk8E;
+    u16 unk90;
+    u16 unk92;
+    s32 unk94;
+} ET_Unk0E;
+
```
Comment (sonicdcer) [inline include/entity.h:150]: can we have struct offset comments?

File: src/st/entity_unk0e.h [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        while (prim != NULL) {
+            prim->blendMode = BLEND_VISIBLE;
+            prim = prim->next;
+        }
+
+        str = self->ext.unk0e.label;
+        self->ext.unk0e.width = *str++;
+        self->ext.unk0e.height = *str++;
+        self->ext.unk0e.label += 2;
+        return;
```
Comment (sonicdcer) [inline src/st/entity_unk0e.h:37]:
```suggestion
break;

File: src/st/entity_unk0e.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    case 0:
+        InitializeEntity(g_InitializeData0);
+        self->flags = (self->flags | FLAG_UNK_10000) ^ FLAG_UNK_08000000;
+        if (self->params == 0) {
+            self->params = 0x60;
+        }
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 3);
+        if (primIndex == -1) {
+            self->step = 0;
+            return;
```
Comment (sonicdcer) [inline src/st/entity_unk0e.h:22]:
```suggestion
break;

File: include/entity.h [hunk 4]
```diff
@@ -461,6 +475,7 @@ typedef union {
     /* 0x7C */ ET_Generic generic;
     /* 0x7C */ ET_EquipItemDrop equipItemDrop;
     /* 0x7C */ ET_RelicOrb relicOrb;
+    /* 0x7C */ ET_Unk0E unk0e;
```
Comment (sonicdcer) [inline include/entity.h:478]:
```suggestion
/* 0x7C */ ET_Unk0E unk0E;

**Review by sonicdcer — APPROVED** [review]

File: include/entity.h (hunk 1 above)
Comment (sonicdcer) [reply include/entity.h:150]:
```suggestion
typedef struct {
    /* 0x7C */ u8* label;
    /* 0x80 */ u16 width;
    /* 0x82 */ u16 height;
    /* 0x84 */ s16 unk84;
    /* 0x86 */ u16 duration;
    /* 0x88 */ u32 unk88;
    /* 0x8C */ s16 unk8C;
    /* 0x8E */ u16 unk8E;
    /* 0x90 */ u16 unk90;
    /* 0x92 */ u16 unk92;
    /* 0x94 */ s32 unk94;
} ET_Unk0E;

File: src/st/entity_unk0e.h (hunk 3 above)
Comment (sonicdcer) [reply src/st/entity_unk0e.h:22]: Sounds reasonable

**Pull request merged**